hack/rpm/tanzu-cli.spec

```diff
@@ -1,13 +1,4 @@
-%if "%{unstable}" == "false"
-Name:       tanzu-cli
-Provides:   tanzu-cli
-Obsoletes:  tanzu-cli  < %{rpm_package_version}
```

Comment:
It turns out that we were using `Provides` and `Obsoletes` wrongly.  Those fields are meant to be used when we rename a package (e.g., `tanzu-cli` -> `newname-cli`) so that the new package can indicate it replaces the old package.  We were not using it correctly.

Those fields do not support using a variable, so removing them allowed me to inject the name of the package.

As a separate change, we could use those fields to allow the `tanzu-cli` package to replace an already installed `tanzu-cli-unstable` package; I tried it and couldn't get it to work quickly, so I left it for later, since it is not relevant to the Centos9 issue.